hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1167,7 +1170,7 @@ public StoreFileWriter createWriterInTmp(long maxKeyCount, Compression.Algorithm
     } else {
       final boolean shouldCacheDataOnWrite = cacheConf.shouldCacheDataOnWrite();
       if (shouldCacheDataOnWrite) {
-        writerCacheConf.enableCacheOnWrite();
+        writerCacheConf.enableCacheOnWriteForCompactions();
```

Comment:
What if I want to cache on write when its other than compactions?

Reply:
Yeah, should be only `enableCacheOnWrite`, as here we are calling it for other writes than compactions.